pkg/controller/v1beta1/inferenceservice/reconcilers/knative/ksvc_reconciler.go

```diff
@@ -86,13 +86,22 @@ func createKnativeService(componentMeta metav1.ObjectMeta,
 		annotations[autoscaling.ClassAnnotationKey] = autoscaling.KPA
 	}
 	lastRolledoutRevision := componentStatus.LatestRolledoutRevision
-	// This is to handle case when the latest ready revision is rolled out with 100% and then rolled back
-	// so here we need to get the revision that is previously rolled out with 100%
-	if componentStatus.LatestRolledoutRevision == componentStatus.LatestReadyRevision &&
-		componentStatus.LatestReadyRevision == componentStatus.LatestCreatedRevision &&
-		componentExtension.CanaryTrafficPercent != nil && *componentExtension.CanaryTrafficPercent < 100 {
+
+	// This is to handle case when the first revision rolled out.
+	// And the canary traffic percent of the first revision should always be 100.
+	if componentStatus.PreviousRolledoutRevision == "" {
 		lastRolledoutRevision = componentStatus.PreviousRolledoutRevision
+		fullCanaryTrafficPercent := int64(100)
+		componentExtension.CanaryTrafficPercent = &fullCanaryTrafficPercent
 	}
+
+	// Log component status and canary traffic percent
+	if componentExtension.CanaryTrafficPercent != nil {
+		log.Info(fmt.Sprintf("LatestRolledoutRevision:%s,LatestReadyRevision:%s,LatestCreatedRevision:%s,PreviousRolledoutRevision:%s,CanaryTrafficPercent:%d", componentStatus.LatestRolledoutRevision, componentStatus.LatestReadyRevision, componentStatus.LatestCreatedRevision, componentStatus.PreviousRolledoutRevision, *componentExtension.CanaryTrafficPercent))
```

Comment:
Here you can use the log key/values pairs pattern used across the repo, so better to be consistent.
```golang
log.Info("revision status:", "LatestRolledoutRevision", componentStatus.LatestRolledoutRevision, "LatestCreatedRevision", componentStatus.LatestCreatedRevision)
```